management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -66,6 +66,23 @@ public synchronized void unregister()
         RpcRegistry.unregister(RPC_CLASS_NAME);
     }
 
+    @Rpc(name = "setFullQuerylog")
+    public void setFullQuerylog(@RpcParam(name="enabled") boolean fullQueryLoggingEnabled) throws UnsupportedOperationException
+    {
+        // Warning: bad design here. Implementations for 3.x adn DSE will throw UnsupportedOperationException at runtime.
```

Comment:
Again, not sure which way is best. Let the 3.11 implementation throw an Exception, or check the version here to cut off the call before it even goes to the agent specific shim.

Reply:
Have a look at the changes I've made. log an error in the server, throw the error in the 3x implementation.